File: GitToolBox/src/main/java/zielu/gittoolbox/ui/config/app/DecorationPartPreview.java
package zielu.gittoolbox.ui.config.app;

import static zielu.gittoolbox.config.DecorationPartType.BRANCH;
import static zielu.gittoolbox.config.DecorationPartType.CHANGED_COUNT;
import static zielu.gittoolbox.config.DecorationPartType.LOCATION;
import static zielu.gittoolbox.config.DecorationPartType.STATUS;
import static zielu.gittoolbox.config.DecorationPartType.TAGS_ON_HEAD;

import com.google.common.collect.ImmutableMap;
import java.util.Map;
import jodd.util.StringBand;
import zielu.gittoolbox.config.DecorationPartType;
import zielu.gittoolbox.ui.StatusPresenter;

class DecorationPartPreview {
  private static final Map<DecorationPartType, String> PREVIEWS = ImmutableMap.of(BRANCH, "master",
      LOCATION, "/path/to/location", TAGS_ON_HEAD, "1.0.0", CHANGED_COUNT, "1 changed");

  private DecorationPartPreview() {
    throw new IllegalStateException();
  }

  static StringBand appendPreview(StatusPresenter presenter, DecorationPartType type, StringBand preview) {
    return preview.append(getPreview(presenter, type));
  }

  private static String getPreview(StatusPresenter presenter, DecorationPartType type) {
    if (type == STATUS) {
      return presenter.aheadBehindStatus(3, 2);
    } else {
      return PREVIEWS.getOrDefault(type, "N/A");
    }
  }
}

File: src/main/kotlin/zielu/gittoolbox/ui/config/v2/DecorationPartPreview.kt
package zielu.gittoolbox.ui.config.v2

import jodd.util.StringBand
import zielu.gittoolbox.config.DecorationPartType
import zielu.gittoolbox.ui.ExtendedRepoInfo
import zielu.gittoolbox.ui.StatusPresenter
import zielu.gittoolbox.util.Count

internal object DecorationPartPreview {
  private val previews = mapOf(
    Pair(DecorationPartType.BRANCH, "master"),
    Pair(DecorationPartType.LOCATION, "/path/to/location"),
    Pair(DecorationPartType.TAGS_ON_HEAD, "1.0.0"),
    Pair(DecorationPartType.CHANGED_COUNT, "1 changed")
  )

  @JvmStatic
  fun appendPreview(presenter: StatusPresenter, type: DecorationPartType, preview: StringBand): StringBand {
    return preview.append(getPreview(presenter, type))
  }

  private fun getPreview(presenter: StatusPresenter, type: DecorationPartType): String {
    return when (type) {
      DecorationPartType.STATUS -> {
        presenter.aheadBehindStatus(3, 2)
      }
      DecorationPartType.CHANGED_COUNT -> {
        presenter.extendedRepoInfo(ExtendedRepoInfo(Count(1)))
      }
      else -> {
        previews.getOrDefault(type, "N/A")
      }
    }
  }
}
